### DISCUSSION TITLE
Specifying custom validation values in Livewire components

### DISCUSSION THREAD

**Discussion by kenlas** [livewire/livewire#4151] [3. Feature ideas]
Hey there,

I was wondering if I could add a `protected $validationCustomValues` property (similar to `$validationAttributes`) to a Livewire component to have the validation values also translated, like here: https://laravel.com/docs/8.x/validation#specifying-values-in-language-files.
I know it can be done inside the global resources/lang/xx/validation.php but it would be more elegant to have these custom values beside the translated attributes in the component.

This could be done in a few lines of code, added to validate() and validateOnly functions like this:
```php
$customValues = $this->getValidationCustomValues();
if (!empty($customValues)) {
    $validator->addCustomValues($customValues);
}
```

After this, one should add this property to the compontent to have the values translated:

```php
protected $validationCustomValues = [
    'payment_type' => [
        'cc' => 'credit card',
    ],
];

```

If you agree, I can make a PR for this.

Thanks,
Keni

**Comment by joshhanley** [comment]
@kenlas makes sense to me! You could also argue that, that is the whole point of the language files though 😬

But if you think it could be useful, then PR's are welcome 🙂

**Reply by kenlas** [reply]
Totally agree with you @joshhanley but then what is the purpose of `$validationAttributes` in the component? 🙂   I think if we have `$validationAttributes` then we could also have `$validationCustomValues` to keep validation-related texts together.
I'll prepare the PR tonight. Thank you!

**Reply by joshhanley** [reply]
@kenlas heh, yeah fair call! 🙂

Ok awesome!
